### PR TITLE
Hkm/ghc924

### PR DESCRIPTION
**Pull request by newhoggy**
This work stream is adding GHC 9.2.4 compatibility _ontop_ of the existing 8.10.7 compatibility. And _only_ that compatibility; there is no intention of changing anything outside of compiler compatibility.

### CODE REVIEW

**Comment by L-as** [comment]
I love you.

File: cabal.project [hunk 1]
```diff
@@ -117,6 +117,7 @@ package ouroboros-consensus-cardano-test
 
 package ouroboros-network
   tests: False
+  ghc-options: -XFlexibleContexts
```
Comment (angerman) [inline cabal.project:120]: @newhoggy I think this needs an explanation. I did see the warning; but are we sure that this is the _right_ fix for ouroboros-network? And does this impact 8.10.7 negatively?

File: nix/haskell.nix [hunk 2]
```diff
@@ -91,9 +94,8 @@ haskell-nix.cabalProject' ({ pkgs
       })
       ({ pkgs, ... }: {
         # Use the VRF fork of libsodium
-        packages = lib.genAttrs [ "cardano-crypto-praos" "cardano-crypto-class" ] (_: {
-          components.library.pkgconfig = lib.mkForce [ [ pkgs.libsodium-vrf ] ];
-        });
+        packages.cardano-crypto-class.components.library.pkgconfig = lib.mkForce [[ pkgs.libsodium-vrf pkgs.secp256k1 ]];
```
Comment (L-as) [inline nix/haskell.nix:97]: Shouldn't this be fixed once and for all in the relevant packages? There's been a lot of work on pkgconfig stuff in haskell.nix lately, cc @hamishmack

File: cabal.project (hunk 1 above)
Comment (newhoggy) [reply cabal.project:132]: This is just temporary to unblock finding all the other compile errors.  The fixes should eventually go into the repository itself.

**Pull request closed without merging**